.github/workflows/ci.yaml

```diff
@@ -39,6 +39,10 @@ jobs:
 
       - run: poetry run invoke ${{ matrix.task }} --check
 
+      - name: Build distributions
+        run: poetry build
```

Comment:
Can you move this to it's own job.
We want these static analysis steps to be fast and only fail if the particular check fails.

Don't want to have the `formatting`/`fmt` step failed because of a build issue.